diesel_derives/src/lib.rs

```diff
@@ -91,7 +91,9 @@ pub fn derive_insertable(input: TokenStream) -> TokenStream {
 
 #[proc_macro_derive(NonAggregate)]
 pub fn derive_non_aggregate(input: TokenStream) -> TokenStream {
-    expand_proc_macro(input, non_aggregate::derive)
+    eprintln!("#[derive(NonAggregate)] is deprecated. Please use \
```

Comment:
Simply putting `#[deprecated]` on the proc macro itself does not work?
And what's about putting it onto the reexport in diesel itself?

Reply:
No, if you put `#[deprecated]` on this function, when you compile `diesel_derives` it will complain about the use of a deprecated function. No output is given on use of the derive. Deprecating a re-export does nothing https://github.com/rust-lang/rust/issues/47236